src/evict/evict_lru.c

```diff
@@ -2898,28 +2898,23 @@ __wti_evict_app_assist_worker(WT_SESSION_IMPL *session, bool busy, bool readonly
         if (!__wt_evict_needed(session, busy, readonly, &pct_full) ||
           (pct_full < 100.0 &&
             (__wt_atomic_loadv64(&evict->eviction_progress) > initial_progress + max_progress)))
-            goto err;
+            break;
 
         if (!__evict_check_user_ok_with_eviction(session, busy))
-            /* At this point ret can only be 0, so it's a clean exit from the loop. */
-            goto err;
+            break;
 
         /* Evict a page. */
```

Comment:
Since this is important code, a brief discussion of why the refactor is correct.

The original case statement handles 5 different situations

1. `ret == 0 && busy` -- break out of loop
2. `ret == 0 && !busy` -- continue loop
3. `ret == WT_NOTFOUND` -- wait, increment counter, continue loop
4. `ret == EBUSY` -- continue loop
5. `ret` is something other than `0`, `WT_NOTFOUND`, or `EBUSY` -- a real error, jump to error label.

By inspection, the new code has the same behavior. But the reviewers should definitely double check this!

Reply:
Would it be okay to break it down by error code like this?

```c
if (ret == 0) {
  if (busy)
    break;
  /* continue the loop if not busy */
} else if (ret == WT_NOTFOUND) {
  ...
} else if (ret != EBUSY) {
  break;
}
```

I think having one type of logic per `if` layer improves readability.

Reply:
Makes sense. Updated in [3f6c69f](https://github.com/wiredtiger/wiredtiger/pull/11495/commits/3f6c69fb8fceb17fd46f3ce3b650e5a24f933d38)